### PR TITLE
WIP - high level scripts testing

### PR DESCRIPTION
**Pull request by frheault**
Creating the next logical step for scripts testing, testing the `-h` test syntax and import/dependencies, this phase of testing simply test if the script can run with simple data.

It does not compare to anything, only if the script can execute with some arguments and input data. This will help to detect if a script uses a function that changed input or output, change of behavior of functions, if supported file format changes, if argument names are modified, or if arguments order changes.

To minimize the size of datasets being fetched, some basic files are downloaded and the processing is performed sequentially to generate files for the next tests.
The only 'mini-pipeline' for now is the small one related to atlas. This is only a preview, everything is ready for 9 mini-pipelines that test 98% of scilpy scripts:
atlas, bst, bundles, connectivity, filtering, others, processing, surface_vtk_fib, tracking, tractometry

EDIT: As mentioned by Arnaud, the use of GDrive for the fetcher could be modified later, right now it is best to quickly adapt the data for the test. This could be done once everything is merged and finished.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Amazing job @frheault I will test this soon but since TRAVIS like it I already like it !

**Comment by GuillaumeTh** [comment]
Thanks @frheault Looks good ! I you want to store on the lab gdrive account the data just day me. I have a question. Why do you correct some scripts that have no link to this PR ?

**Comment by frheault** [comment]
@GuillaumeTh All of my tests are ready for all scripts, so I fixed everything to make the tests run. BUT I decided to start with an introduction to the tests to make discussion and review easier. Sadly these fixes were a bit everywhere in many commit so I can't cherry pick.

File: scilpy/io/fetcher.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    ValueError
+        Raises if the md5 checksum of the file does not match the expected
+        value. The downloaded file is not deleted when this error is raised.
+    """
+    scilpy_home = get_home()
+
+    if not os.path.exists(scilpy_home):
+        os.makedirs(scilpy_home)
+
+    to_dezip = {}
```
Comment (jchoude) [inline scilpy/io/fetcher.py:114]: `dezip` -> `unzip`

**Comment by jchoude** [comment]
GJ! At some point, we might want to refactor this to have some class system to avoid always having to  write the beginning of the test script (data fetching, etc)

**Pull request merged**